R/uvhydrograph-data.R

```diff
@@ -215,27 +215,120 @@ parseUVSupplemental <- function(data, plotName, pts, useDownsampled=FALSE) {
 correctionsTable <- function(data) {
   if (any(names(data) %in% c("series_corr", "series_corr_ref", "series_corr_up", "series_corr2"))) {
     corrections <- data[[grep("series_corr", names(data))]]
-    corrections_table <- as.data.frame(cbind(seq(nrow(corrections)), corrections$comment))
-    colnames(corrections_table) <- c("", "Comments")
+    corrections_table <- as.data.frame(cbind(seq(nrow(corrections)), as.character(corrections$time), corrections$comment))
+    colnames(corrections_table) <- c("", "Time", "Comments")
     return(corrections_table)
   } else (return(corrections_table <- NULL))
 }
 
+addGroupCol <- function(data, newColumnName, isNewCol, newGroupValue=NULL, groupChildValue=NULL, vars=NULL){
+  build_vec <- c()
+  prev <- NULL
+  for(r in 1:nrow(data)){
```

Comment:
I think this gets at what we discussed yesterday when talking about loops vs apply statements for efficiency. We are looping over the data, but this data is only comments so it should have less than 100 (presumably). So in this case, I think using the loop is fine. Just something to make note of for the future.